cmd/root.go

```diff
@@ -0,0 +1,89 @@
+package main
+
+import (
+	"encoding/json"
+	"log"
+	"os"
+	"time"
+
+	"github.com/spec-tacles/spectacles.go/broker"
+	"github.com/spec-tacles/spectacles.go/gateway"
+	"github.com/spec-tacles/spectacles.go/rest"
+	"github.com/spec-tacles/spectacles.go/types"
+	"github.com/spf13/cobra"
+)
+
+var amqpUrl, amqpGroup, token, logLevel string
+var shardCount int
+
+var brokerConnected bool
+
+var logger = log.New(os.Stdout, "[CMD] ", log.Ldate|log.Ltime|log.Lshortfile)
+var logLevels = map[string]int{
+	"suppress": gateway.LogLevelSuppress,
+	"info":     gateway.LogLevelInfo,
+	"warn":     gateway.LogLevelWarn,
+	"debug":    gateway.LogLevelDebug,
+	"error":    gateway.LogLevelError,
+}
+
+func main() {
+	rootCmd.Execute()
+}
+
+var rootCmd = &cobra.Command{
+	Use:   "spectacles",
+	Short: "Connects to the Discord websocket API using spectacles.go",
+	Run: func(cmd *cobra.Command, args []string) {
+		amqp := broker.NewAMQP(amqpGroup, "", func(string, []byte) {})
+		go tryConnect(amqp)
```

Comment:
It might be worthwhile to do this synchronously to avoid dropping messages from Discord in the event that we're unable to connect to the AMQP server immediately. Dropping messages is a big bad since there is a lot of potentially critical data that Discord sends within the first few hundred milliseconds of a connection (specifically, ready packets and guild creates).